server/app/controllers/admin/AdminApplicationController.java

```diff
@@ -457,12 +464,24 @@ public Result index(
         applicationListView.render(
             request,
             program,
+            getAllApplicationStatusesForProgram(program),
             paginationSpec,
             applications,
             RenderFilterParams.builder()
                 .setSearch(search)
                 .setFromDate(fromDate)
                 .setUntilDate(untilDate)
+                .setSelectedApplicationStatus(applicationStatus)
                 .build()));
   }
+
+  private ImmutableList<String> getAllApplicationStatusesForProgram(ProgramDefinition program) {
```

Comment:
Done. My general preference is to use a strong type instead of a database ID since I think it conveys more meaning. That said, in this case, we're actually calling a method that ends up being for all versions of the program, not just the one for the given ID, so it probably was ending up being more confusing than helpful.